lib/xclarity_client/services/mixins/response_builder_mixin.rb

```diff
@@ -33,6 +33,27 @@ def build_response_with_resource_list(response, endpoint_class)
           endpoint_class.new resource_params
         end
       end
+
+      def build_response_with_opts(resource, uri, opts)
```

Comment:
This method seems to do the exactly same thing that `connection.do_get`. Maybe this code could replace this method:
```ruby
query = opts.reject {|key| key.eql?('headers')}
headers = opts['headers']

connection.do_get(uri, query, headers)
```
If a validation in `opts` content is needed, then you can replace the first line of the code above with:
```ruby
allowed_query_keys = ['type', 'sort']
query = opts.select {|key| allowed_query_keys.include?(key)}
```
Also, once the responsibility of this mixin is build responses, and not requests (which is what this method is doing), it would be good to move this code to another place, maybe to `XClarityClient::Services::XClarityService#get_response_with_opts`

What do you think?

Reply:
@douglasgabriel This is actually 10 times better hahaha. I fixed it, so it should work fine with this.